### PR TITLE
Avoid commit errors and fail to disconnect from the database.

### PR DESCRIPTION
**Pull request by Z-2020**
For example, when the database is set up with optimistic lock,
the commit will give an error and the connection cannot be broken.

### CODE REVIEW

File: databases/core.py [hunk 1]
```diff
@@ -363,7 +364,11 @@ async def commit(self) -> None:
         async with self._connection._transaction_lock:
             assert self._connection._transaction_stack[-1] is self
             self._connection._transaction_stack.pop()
-            await self._transaction.commit()
+            try:
+                await self._transaction.commit()
```
Comment (vmarkovtsev) [inline databases/core.py:368]: (just a question) will the following work?
```python
try:
    transaction.commit()
except IntegrityError:
    transaction.rollback()
```
That is, catching it in the user code instead.

Comment (Z-2020) [reply databases/core.py:368]: > (just a question) will the following work?
> 
> ```python
> try:
>     transaction.commit()
> except IntegrityError:
>     transaction.rollback()
> ```
> 
> That is, catching it in the user code instead.

async def **rollback**(self) -> None:
        async with self._connection._transaction_lock:
            assert **self._connection._transaction_stack[-1]** is self  # Here _transaction_stack will be an empty list.
           ......

when I use **transaction.start()** before, unfortunately, TIDB did not support SAVEPOINT.
May I ask if you have any other suggestions? 

I look forward to your reply!

**Comment by vmarkovtsev** [comment]
@Z-2020 can I close this?

**Comment by vmarkovtsev** [comment]
Feel free to ping me if there is something left.

**Pull request closed without merging**